### PR TITLE
Move to MongoDB.Entities

### PR DESCRIPTION
**Pull request by JustRoxy**

### CODE REVIEW

File: backend/CircleForms/CircleForms.csproj [hunk 1]
```diff
@@ -18,6 +18,7 @@
         <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="6.0.1" />
         <PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.10.9" />
         <PackageReference Include="MongoDB.Driver" Version="2.13.2" />
+        <PackageReference Include="MongoDB.Entities" Version="21.0.0-alpha3" />
```
Comment (stanriders) [inline backend/CircleForms/CircleForms.csproj:21]: Do we want to use alpha? I'd prefer a stable version if we can use it

File: backend/CircleForms/Controllers/UsersController.cs [hunk 2]
```diff
@@ -24,13 +24,13 @@ public UsersController(ILogger<UsersController> logger, IUserRepository usersSer
...
-    /// Get user data. (Requires auth, Requires Admin role)
+    ///     Get user data. (Requires auth, Requires Admin role)
     /// </summary>
     [Authorize(Roles = "Admin")]
-    [HttpGet("{id:long}")]
+    [HttpGet("{id}")]
     [ProducesResponseType(typeof(User), StatusCodes.Status200OK, "application/json")]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
-    public async Task<IActionResult> Get(long id)
+    public async Task<IActionResult> Get(string id)
```
Comment (stanriders) [inline backend/CircleForms/Controllers/UsersController.cs:33]: We might wanna exit early if id isn't a long. Could probably use 
````CSharp 
public async Task<IActionResult> Get([RegularExpression(@"^\d$")] string id)
````

File: backend/CircleForms/Controllers/UsersController.cs [hunk 3]
```diff
@@ -56,11 +56,11 @@ public async Task<List<User>> GetAll()
     }
 
     /// <summary>
-    /// Set user role. (Requires auth, Requires SuperAdmin role)
+    ///     Set user role. (Requires auth, Requires SuperAdmin role)
     /// </summary>
     [Authorize(Roles = "SuperAdmin")]
     [HttpPatch]
-    public async Task<User> EscalatePrivileges(long id, int role)
+    public async Task<User> EscalatePrivileges(string id, int role)
```
Comment (stanriders) [inline backend/CircleForms/Controllers/UsersController.cs:63]: Same as above

File: backend/CircleForms/Models/Posts/Post.cs [hunk 4]
```diff
@@ -58,4 +49,15 @@ public Post(string id)
...
+    public string GenerateNewID()
+    {
+        return ObjectId.GenerateNewId().ToString();
+    }
+
+    [ObjectId]
+    [BsonId]
+    [JsonProperty("id")]
+    [SwaggerSchema(ReadOnly = true)]
+    public string ID { get; set; }
```
Comment (stanriders) [inline backend/CircleForms/Models/Posts/Post.cs:62]: Does it have to be full uppercase? Pretty sure that's against the naming convention

**Review by stanriders — CHANGES_REQUESTED** [review]

File: backend/CircleForms/Models/Posts/Post.cs (hunk 4 above)
Comment (JustRoxy) [reply backend/CircleForms/Models/Posts/Post.cs:62]: yep

Comment (stanriders) [reply backend/CircleForms/Models/Posts/Post.cs:62]: You should add it to the naming exceptions then so that we wont get squiggly lines all the time

Comment (JustRoxy) [reply backend/CircleForms/Models/Posts/Post.cs:62]: fix it in your ide of choice

**Review by stanriders — APPROVED** [review]

**Pull request merged**